integration-tests/src/frame/balances.rs

```diff
@@ -95,6 +97,87 @@ async fn tx_basic_transfer() -> Result<(), subxt::Error<DispatchError>> {
     Ok(())
 }
 
+#[tokio::test]
+async fn dry_run_passes() -> Result<(), subxt::Error<DispatchError>> {
+    let alice = pair_signer(AccountKeyring::Alice.pair());
+    let bob = pair_signer(AccountKeyring::Bob.pair());
+    let bob_address = bob.account_id().clone().into();
+    let cxt = test_context().await;
+    let api = &cxt.api;
+
+    let signed_extrinsic = api
+        .tx()
+        .balances()
+        .transfer(bob_address, 10_000)?
+        .create_signed(&alice, Default::default())
+        .await?;
+
```

Comment:
This test fails unless a `sleep` is added on this line: `tokio::time::sleep(Duration::from_secs(2)).await;`

I'm not sure how to investigate why this might happen, do you have any suggestions? The error is: `TransactionValidityError::Invalid(InvalidTransaction::BadProof`

Reply:
I wonder whether adding a sleep above the `let signed_extrinsic` line also works?

(I'm wondering whether it's something like the test_context needs to produce a block first or something before it accepts the extrinsic?!)

Reply:
Managed to fix it on my local, submitting a prior extrinsic seemed to somehow interfere with the dryrun rpc so I removed the extrinsic.